### PR TITLE
Feat: Delete Pin Route Backend

### LINKED ISSUE #24: Pin Deletion Route

### PR DESCRIPTION
**Pull request by KGsxj**
#24

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "prepare": "husky install",
```
Comment (brarsanmol) [inline api/package.json]: revert: do not modify this.

File: api/src/pin/pin.controller.ts [hunk 2]
```diff
@@ -21,6 +21,36 @@ class PinController {
       ? response.status(200).send(dto)
       : response.status(404).send(dto);
   }
+
+  @Delete('/:id') 
+    async deleteByID(@Response() response: ExpressResponse, id: string) {
```
Comment (brarsanmol) [inline api/src/pin/pin.controller.ts]: nit: rename function to deleteByid not deleteByID

File: api/src/pin/pin.controller.ts [hunk 3]
```diff
@@ -21,6 +21,36 @@ class PinController {
...
+  //   const dto: PinResponseDto = result
+  //     ? ({ status: 'Success', pin: result } as PinResponseDto)
+  //     : ({ status: 'Error '} as PinResponseDto)
+
+  //   return result
+  //   ? response.status(200).send(dto)
+  //   : response.status(404).send(dto);
+  // }
+  
+
```
Comment (brarsanmol) [inline api/src/pin/pin.controller.ts]: fix: delete this additional code.

File: api/src/pin/pin.service.ts [hunk 4]
```diff
@@ -8,6 +8,34 @@ class PinService {
...
     return await this.prisma.pin.findMany();
   }
+
+  // async findUnique(pin_ID: string) {
+  //   return await this.prisma.pin.findUnique({
+  //     where: {
+  //       name: pin_ID
+  //     }
+  //   })
+  // }
```
Comment (brarsanmol) [inline api/src/pin/pin.service.ts]: fix: delete this commented code.

File: api/src/pin/pin.service.ts [hunk 5]
```diff
@@ -8,6 +8,34 @@ class PinService {
...
+
+    // Delete a Pin.
+    // @param {PinDeleteArgs} args - Arguments to delete one Pin.
+    // @example
+    // // Delete one Pin
+    // const Pin = await prisma.pin.delete({
+    // where: {
+    // // ... filter to delete one Pin
+    // }
+    // })
```
Comment (brarsanmol) [inline api/src/pin/pin.service.ts]: fix: delete this commented code.

**Review by brarsanmol — CHANGES_REQUESTED** [review]
good work, some small style and nit's need to be fixed.

**Review by brarsanmol — APPROVED** [review]
LGTM!

**Review by yoyopyeah — APPROVED** [review]

**Pull request merged**